File: 8-1.py
#!/usr/bin/env python

lines = [l.split('|')[1] for l in open('8.input').readlines()]
lines = [l.split() for l in lines]
lines = [item for sublist in lines for item in sublist]

count = 0
for s in lines:
    if len(s) == 2:
        count += 1
    elif len(s) == 4:
        count += 1
    elif len(s) == 3:
        count += 1
    elif len(s) == 7:
        count += 1

print(count)

File: 8-1.rb
#!/usr/bin/env ruby

lines = File.read('8.input').lines.map { |l| l.split '|' }.map { |l| l[1].split }

print lines.flatten.count { |s|
  case s.size
  when 2
    true
  when 4
    true
  when 3
    true
  when 7
    true
  else
    false
  end
}
print "\n"
